app/controllers/admin/sessions_controller.rb

```diff
@@ -0,0 +1,28 @@
+class Admin::SessionsController < AdminApplicationController
+  before_action :authenticate_user!, only: %i[logout]
+  before_action :prevent_access_for_authenticated_user!, only: %i[login]
+
+  def login
+    if request.post?
+      login = Login.authenticate(params[:email], params[:password])
+      if login
+        session[:login_id] = login.id
+        session[:expires_at] = Time.current + 24.hour
+        if login.user.administrator?
+          redirect_to '/' and return
+        else
+          clear_session
+          render json: { message: "Missing administrator privileges" }, status: :unauthorized and return
+        end
+      else
+        render json: { message: "Invalid credentials" }, status: :unauthorized and return
+      end
+    end
+    render 'user/sessions/login'
```

Comment:
Use redirect_to instead.

Reply:
That would run the controller in the user portal as well, which wouldn't redirect back to this portal. Redirecting between portals isn't straightforward anyway because they live on separate domains. I don't see how I can help this.